metadata-ingestion/src/datahub/ingestion/source/unity/source.py

```diff
@@ -309,9 +309,19 @@ def get_workunits_internal(self) -> Iterable[MetadataWorkUnit]:
             )
 
         if self.config.is_profiling_enabled():
-            self.report.report_ingestion_stage_start("Wait on warehouse")
-            assert wait_on_warehouse
-            wait_on_warehouse.result()
+            self.report.report_ingestion_stage_start("Start warehouse")
+            # Can take several minutes, so start now and wait later
+            wait_on_warehouse = self.unity_catalog_api_proxy.start_warehouse()
+            if wait_on_warehouse is None:
+                self.report.report_failure(
+                    "initialization",
+                    f"SQL warehouse {self.config.profiling.warehouse_id} not found",
```

Comment:
So this happens when permission for sql warehouse is missing Or incorrect sql warehouse is specified ?

Reply:
I need to check, as this was the existing codebase

Reply:
this happens when there is no warehouse with specified id.

Reply:
makes sense